Python 3.10: TypeError: PurePath._from_parts() got an unexpected keyword argument 'init'

Hello, in Fedora we see test failures with Python 3.10.0b1:

```
...
______________________ TestCWD.test_cwd_root_dir_pathlike ______________________
self = <tests.test_glob.TestCWD object at 0x7f4b20972bf0>
    def test_cwd_root_dir_pathlike(self):
        """Test root level glob when we switch directory via `root_dir` with a path-like object."""
    
>       self.assert_equal(glob.glob('EF', root_dir=pathlib.Path(self.tempdir)), ['EF'])
tests/test_glob.py:1136: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
cls = <class 'wcmatch.pathlib.PosixPath'>, args = ('@test_188_tmp_dir',)
kwargs = {}
    def __new__(cls, *args, **kwargs):
        """New."""
    
        if cls is Path:
            cls = WindowsPath if os.name == 'nt' else PosixPath
>       self = cls._from_parts(args, init=False)
E       TypeError: PurePath._from_parts() got an unexpected keyword argument 'init'
wcmatch/pathlib.py:80: TypeError
...
FAILED tests/test_glob.py::TestCWD::test_cwd_root_dir_pathlike - TypeError: P...
FAILED tests/test_globmatch.py::TestGlobMatchSpecial::test_filter_root_dir_pathlib
FAILED tests/test_globmatch.py::TestGlobMatchSpecial::test_match_root_dir_pathlib
FAILED tests/test_pathlib.py::TestGlob::test_glob - TypeError: PurePath._from...
FAILED tests/test_pathlib.py::TestGlob::test_integrity - TypeError: PurePath....
FAILED tests/test_pathlib.py::TestGlob::test_relative - TypeError: PurePath._...
FAILED tests/test_pathlib.py::TestGlob::test_relative_exclude - TypeError: Pu...
FAILED tests/test_pathlib.py::TestGlob::test_rglob - TypeError: PurePath._fro...
FAILED tests/test_pathlib.py::TestPathlibGlobmatch::test_cases[case0] - TypeE...
FAILED tests/test_pathlib.py::TestPathlibGlobmatch::test_cases[case1] - TypeE...
FAILED tests/test_pathlib.py::TestPathlibGlobmatch::test_cases[case2] - TypeE...
FAILED tests/test_pathlib.py::TestPathlibGlobmatch::test_cases[case3] - TypeE...
FAILED tests/test_pathlib.py::TestPathlibGlobmatch::test_cases[case4] - TypeE...
FAILED tests/test_pathlib.py::TestPathlibGlobmatch::test_cases[case5] - TypeE...
FAILED tests/test_pathlib.py::TestPathlibMatch::test_cases[case0] - TypeError...
FAILED tests/test_pathlib.py::TestPathlibMatch::test_cases[case1] - TypeError...
FAILED tests/test_pathlib.py::TestPathlibMatch::test_cases[case2] - TypeError...
FAILED tests/test_pathlib.py::TestPathlibMatch::test_cases[case3] - TypeError...
FAILED tests/test_pathlib.py::TestPathlibMatch::test_cases[case4] - TypeError...
FAILED tests/test_pathlib.py::TestPathlibMatch::test_cases[case5] - TypeError...
FAILED tests/test_pathlib.py::TestExceptions::test_absolute_glob - TypeError:...
FAILED tests/test_pathlib.py::TestExceptions::test_bad_path - TypeError: Pure...
FAILED tests/test_pathlib.py::TestExceptions::test_inverse_absolute_glob - Ty...
FAILED tests/test_pathlib.py::TestComparisons::test_equal - TypeError: PurePa...
FAILED tests/test_pathlib.py::TestComparisons::test_flavour_equal - TypeError...
FAILED tests/test_pathlib.py::TestComparisons::test_instance - TypeError: Pur...
FAILED tests/test_pathlib.py::TestComparisons::test_pickle - TypeError: PureP...
FAILED tests/test_pathlib.py::TestExpansionLimit::test_limit_glob - TypeError...
FAILED tests/test_pathlib.py::TestExpansionLimit::test_limit_rglob - TypeErro...
========== 29 failed, 1112 passed, 148 skipped, 1 deselected in 4.35s ==========

```

See https://bugs.python.org/issue40038 and https://github.com/python/cpython/commit/2219187cab6bca009c42b63b2f4c30b5b10c916d

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -41,15 +41,9 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - name: Set up Python ${{ matrix.python-version }}
-      # if: "!endsWith(matrix.python-version, '-dev')"
       uses: actions/setup-python@v2
       with:
         python-version: ${{ matrix.python-version }}
-    # - name: Set up development Python ${{ matrix.python-version }}
-    #   if: endsWith(matrix.python-version, '-dev')
-    #   uses: deadsnakes/action@v1.0.0
-    #   with:
-    #     python-version: ${{ matrix.python-version }}
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip setuptools tox coverage codecov
@@ -79,7 +73,7 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v2
       with:
         python-version: ${{ matrix.python-version }}
     - name: Install dependencies
@@ -103,7 +97,7 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v2
       with:
         python-version: ${{ matrix.python-version }}
     - name: Install dependencies

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -17,7 +17,7 @@ jobs:
       max-parallel: 4
       matrix:
         platform: [ubuntu-latest, windows-latest]
-        python-version: [3.6, 3.7, 3.8, 3.9]
+        python-version: [3.6, 3.7, 3.8, 3.9, 3.10.0-beta.1]
         include:
           - python-version: 3.6
             tox-env: py36
@@ -27,6 +27,11 @@ jobs:
             tox-env: py38
           - python-version: 3.9
             tox-env: py39
+          - python-version: 3.10.0-beta.1
+            tox-env: py310
+        exclude:
+          - platform: windows-latest
+            python-version: 3.10.0-beta.1
 
     env:
       TOXENV: ${{ matrix.tox-env }}
@@ -36,15 +41,15 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - name: Set up Python ${{ matrix.python-version }}
-      if: "!endsWith(matrix.python-version, '-dev')"
-      uses: actions/setup-python@v1
+      # if: "!endsWith(matrix.python-version, '-dev')"
+      uses: actions/setup-python@v2
       with:
         python-version: ${{ matrix.python-version }}
-    - name: Set up development Python ${{ matrix.python-version }}
-      if: endsWith(matrix.python-version, '-dev')
-      uses: deadsnakes/action@v1.0.0
-      with:
-        python-version: ${{ matrix.python-version }}
+    # - name: Set up development Python ${{ matrix.python-version }}
+    #   if: endsWith(matrix.python-version, '-dev')
+    #   uses: deadsnakes/action@v1.0.0
+    #   with:
+    #     python-version: ${{ matrix.python-version }}
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip setuptools tox coverage codecov

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -20,7 +20,7 @@ jobs:
       with:
         fetch-depth: 0
     - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v2
       with:
         python-version: ${{ matrix.python-version }}
     - name: Install dependencies
@@ -45,7 +45,7 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
-      - uses: actions/setup-python@v1
+      - uses: actions/setup-python@v2
         with:
           python-version: 3.9
       - name: Package

--- a/tox.ini
+++ b/tox.ini
@@ -1,7 +1,7 @@
 [tox]
 skipsdist=true
 envlist=
-    py36,py37,py38,py39,
+    py36,py37,py38,py39,py310,
     lint
 
 [testenv]

--- a/wcmatch/pathlib.py
+++ b/wcmatch/pathlib.py
@@ -3,6 +3,7 @@ import pathlib
 import os
 from . import glob
 from . import _wcparse
+from . import util
 
 __all__ = (
     "CASE", "IGNORECASE", "RAWCHARS", "DOTGLOB", "DOTMATCH",
@@ -77,10 +78,14 @@ class Path(pathlib.Path):
 
         if cls is Path:
             cls = WindowsPath if os.name == 'nt' else PosixPath
-        self = cls._from_parts(args, init=False)
+        if util.PY310:  # pragma: no cover
+            self = cls._from_parts(args)
+        else:
+            self = cls._from_parts(args, init=False)
         if not self._flavour.is_supported:
             raise NotImplementedError("Cannot instantiate {!r} on your system".format(cls.__name__))
-        self._init()
+        if not util.PY310:
+            self._init()
         return self
 
     def glob(self, patterns, *, flags=0, limit=_wcparse.PATTERN_LIMIT):

--- a/wcmatch/util.py
+++ b/wcmatch/util.py
@@ -9,6 +9,7 @@ from functools import wraps
 import warnings
 
 PY37 = (3, 7) <= sys.version_info
+PY310 = (3, 10) <= sys.version_info
 
 UNICODE = 0
 BYTES = 1

